_posts/2020-08-15-Named-Graphs.md

```diff
@@ -0,0 +1,539 @@
+access to quads from a Named Graph but the Default Graph is always
+available outside the `graph` keyword.  For instance:
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a
+from <http://example.com/Rawl_B502>
+from named <http://example.com/LL>
+where {
+    graph ?g {
+        ?a irishRel:nomName "Find"
+    }
+}
+```
+
+The query above will again only return 19 results, which is because it
+does not search the Default Graph.
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a ?g
+from <http://example.com/Rawl_B502>
+from named <http://example.com/LL>
+where {
+    { ?a irishRel:nomName "Find" }
+    union {
+		graph ?g {
+			?a irishRel:nomName "Find"
+		}
+    }	
+}
+```
+
+This illustrative query will return 21 results where genealogical
+quads from [Rawl
+B502](https://,en.wikipedia.org/wiki/Bodleian_Library,_MS_Rawlinson_B_502)
+is pulled into the default graph and searched while information from
+LL stays within its Named Graph and is searched separately.  The
+combination of both graphs is merged using the `union` SPARQL keyword
+which merges two result sets into a single result set.  However, see
+below concerning how Stardog's and GraphDB's `owl:sameAs` semantics can
+make these results confusing in different ways depending on the
+Triplestore used.  The above query can be written just the same using
+two `from named` statements and using just the `graph` keyword.  As an
+example:
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a ?g
+from named <http://example.com/Rawl_B502>
+from named <http://example.com/LL>
+where {
+	graph ?g {
+		?a irishRel:nomName "Find"
+	}	
+}
+```
+
+The last question that may be asked is: how does a user query all
+graphs without needing to specify each?  This is not currently
+possible in the [SPARQL 1.1
+standard](https://www.w3.org/TR/sparql11-overview/) so this is where
+the situation becomes complicated and the solution depends on the
+Triplestore that the user chooses.  In Stardog there are two ways to
+do this.  First there is a special graph named
+`<tag:stardog:api:context:all>`.  This special graph, only available
+in Stardog, will automatically pull Named Graph all quads into the
+Default Graph.  Second, the user can set a property on the database in
+the Stardog settings named `Query All Graphs` which pulls all Named
+Graph quads into the Default Graph by default, which makes using the
+`from named`, `from`, and `graph` keywords all unnecessary to search
+all quads at once.
+
+
+Turning to GraphDB, the situation is slightly different.  Taking the
+original naive query from above:
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?a
+where {
+    ?a irishRel:nomName "Find"
+}
+```
+
+In GraphDB, this query will return 105 results.  Why does this return
+so many results?  This is due to the way in which GraphDB constructs
```

Comment:
I think you should mention somewhere that 105 is the actual total number of individuals called "Find" in the database. The issue is why GraphDB is returning the overall total by default. The way this is phrased, the reader could be left wondering if this result is somehow in error.